src/cosl/coordinated_workers/worker.py

```diff
@@ -316,6 +322,11 @@ def check_readiness(self) -> ServiceEndpointStatus:
             logger.exception("Unexpected exception getting readiness endpoint")
         return ServiceEndpointStatus.down
 
+    @property
+    def is_tls_enabled(self) -> bool:
+        """Check if the worker has the TLS certificate on disk."""
+        return self._container.exists(CERT_FILE)
```

Comment:
I think this check is a bit too naive. What if we're handling a certificates-relation-broken? what if there was an issue with deleting the cert and it's still there but invalid? 
compare this with [traefik's method](https://github.com/canonical/traefik-k8s-operator/blob/main/src/charm.py#L367) and [tempo-k8s](https://github.com/canonical/tempo-k8s-operator/blob/main/src/charm.py#L177). why the difference? which one is correct?

(btw, we really should consider generalizing this and/or agreeing on what is **the way** to check for tls readiness)

Reply:
Tempo and traefik both have acesss to Cert Handler, the worker does not; we can't do the check the same way; the only other way is to check if the coordinator has put the cert in relation data